lib/SILOptimizer/Mandatory/Differentiation.cpp

```diff
@@ -3530,7 +3530,8 @@ class AdjointEmitter final : public SILInstructionVisitor<AdjointEmitter> {
       localAllocBuilder.setInsertionPoint(
           getAdjoint().getEntryBlock()->begin());
     else
-      localAllocBuilder.setInsertionPoint(localAllocations.back());
+      localAllocBuilder.setInsertionPoint(
```

Comment:
It would be nice not to set the insertion point in this `else` branch every time, but unfortunately it's necessary to avoid `stack dealloc does not match most recent stack alloc` verification failure.

I believe this is because the insertion point of `localAllocBuilder` gets moved back as `builder` creates instructions, so the insertion point is no longer at the top of the adjoint entry BB.